includes/functions.php

```diff
@@ -2427,8 +2427,11 @@ function pmpro_getLevelAtCheckout( $level_id = null, $discount_code = null ) {
 		$level_id = apply_filters( 'pmpro_default_level', intval( $default_level ) );
 	}
 
+	// Cast level ID to int to be safe.
+	$level_id = (int) $level_id;
+
 	// If we still don't have a level, bail.
-	if ( empty( $level_id ) || $level_id < 1 || ! is_int( $level_id ) ) {
+	if ( empty( $level_id ) || $level_id < 1 ) {
```

Comment:
Remove the cast and comment above.

Even though something like "4" < 1 still evaluates to false and "0" < 1 evaluates to true in the versions of PHP I tested, it feels safe to future proof this against any strict typing coming down the pipe in PHP by casting to an in here in the check.

https://3v4l.org/NPn5Z

`if ( empty( $level_id ) || (int)$level_id < 1 ) {`